### PR TITLE
For #5662: Implement awesomebar related telemetry.

### PR DESCRIPTION
**Pull request by mcarare**
# Request for data collection review form

**All questions are mandatory. You must receive a review from a data steward peer on your responses to these questions before shipping new data collection.**

_1) What questions will you answer with this data?_
How often do users choose items from search suggestions?

_2) Why does Mozilla need to answer these questions?  Are there benefits for users? Do we need this information to address product or business requirements?_
This is needed to analyze feature usage and possible feature improvements.

_3) What alternative methods did you consider to answer these questions? Why were they not sufficient?_
This is the only way.

_4) Can current instrumentation answer these questions?_
No. We need a new specific event for a specific app feature.

_5) List all proposed measurements and indicate the category of data collection for each measurement, using the [Firefox data collection categories](https://wiki.mozilla.org/Firefox/Data_Collection) found on the Mozilla wiki._

_**Note that the data steward reviewing your request will characterize your data collection based on the highest (and most sensitive) category.**_

<table>
  <tr>
    <td>Measurement Description</td>
    <td>Data Collection Category</td>
    <td>Tracking Bug #</td>
  </tr>

  <tr>
    <td>Search Suggestion tapped</td>
    <td>Category 2</td>
    <td>https://github.com/mozilla-mobile/focus-android/issues/5662 </td>
    </tr>

  <tr>
    <td>Typed text search tapped</td>
    <td>Category 2</td>
    <td>https://github.com/mozilla-mobile/focus-android/issues/5662 </td>
    </tr>

  <tr>
    <td>autocomplete arrow tapped</td>
    <td>Category 2</td>
    <td>https://github.com/mozilla-mobile/focus-android/issues/5662 </td>
    </tr>

</table>

_6) How long will this data be collected?  Choose one of the following:_
    Expiry for capturing data is set to "2022-11-01"

_7) What populations will you measure?_
All release channels and locales.

_8) If this data collection is default on, what is the opt-out mechanism for users?_
 Users can opt out of data collection by disabling Usage and technical data from Settings -> Privacy and security -> Data choices.

_9) Please provide a general description of how you will analyze this data._
 Glean 

_10) Where do you intend to share the results of your analysis?_
Only on Glean, mobile teams, and BD have internal access.

_12) Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection?_
No.

### CODE REVIEW

File: app/metrics.yaml [hunk 1]
```diff
@@ -1249,3 +1249,29 @@ context_menu:
...
+    description: Search suggestion selected.
+    bugs:
+      - https://github.com/mozilla-mobile/focus-android/issues/5662
+    data_reviews:
+      - https://github.com/mozilla-mobile/focus-android/pull/?
+    data_sensitivity:
+      - interaction
+    notification_emails:
+      - android-probes@mozilla.com
+    expires: "2022-11-01"
```
Comment (st3fan) [inline app/metrics.yaml:1265]: @mcarare I think we want the same `engine_name` extra here that we have in `search_bar.performed_search` right?

```
    extra_keys:
      engine_name:
        description: The name of the engine used to perform the search.
        type: string
```

**Review by st3fan — CHANGES_REQUESTED** [review]

File: app/metrics.yaml (hunk 1 above)
Comment (st3fan) [reply app/metrics.yaml:1265]: Or does this trigger a search event later that does have the engine_name in the event? If that is the case then this is probably fine.

File: app/metrics.yaml [hunk 2]
```diff
@@ -1249,3 +1249,29 @@ context_menu:
           open_in_new_tab, open_in_private_tab, open_image_in_new_tab,
           save_image, share_link, copy_link, copy_image_location, share_image
         type: string
+
+search_suggestions:
+  suggestion_tapped:
```
Comment (st3fan) [inline app/metrics.yaml:1254]: For iOS I also added this:

```
  search_tapped:
    type: event
    description: The original query is selected.
    bugs:
      - https://github.com/mozilla-mobile/focus-android/issues/5662
    data_reviews:
      - https://github.com/mozilla-mobile/focus-android/pull/?
    data_sensitivity:
      - interaction
    notification_emails:
      - android-probes@mozilla.com
    expires: "2022-11-01"
```

THis covers the case where the user taps the top entry, which is simply the thing they typed and not a suggestion that came back from the search provider.

File: app/metrics.yaml (hunk 1 above)
Comment (mcarare) [reply app/metrics.yaml:1364]: It does not trigger the other event, as that is specific to another action. 
I added the engine name as an extra key.

File: app/metrics.yaml (hunk 2 above)
Comment (mcarare) [reply app/metrics.yaml:1353]: In Focus the first item is still a suggestion. I can compare the value to the actual typed text if this differentiation helps.

**Comment by mcarare** [comment]
@st3fan Thank you for your suggestions. I made some changes to the code.

File: app/metrics.yaml (hunk 1 above)
Comment (st3fan) [reply app/metrics.yaml:1364]: I'll do the same on iOS.

**Comment by travis79** [comment]
> # Request for data collection review form
> 
> **All questions are mandatory. You must receive a review from a data steward peer on your responses to these questions before shipping new data collection.**
> 
> _1) What questions will you answer with this data?_ How often do users choose items from search suggestions?
> 
> _2) Why does Mozilla need to answer these questions? Are there benefits for users? Do we need this information to address product or business requirements?_ This is needed to analyze feature usage and possible feature improvements.
> 
> _3) What alternative methods did you consider to answer these questions? Why were they not sufficient?_ This is the only way.
> 
> _4) Can current instrumentation answer these questions?_ No. We need a new specific event for a specific app feature.
> 
> _5) List all proposed measurements and indicate the category of data collection for each measurement, using the [Firefox data collection categories](https://wiki.mozilla.org/Firefox/Data_Collection) found on the Mozilla wiki._
> 
> _**Note that the data steward reviewing your request will characterize your data collection based on the highest (and most sensitive) category.**_
> Measurement Description 	Data Collection Category 	Tracking Bug #
> Search Suggestion tapped 	Category 2 	https://github.com/[/issues/5662](https://github.com/mozilla-mobile/focus-android/issues/5662)
> Typed text search tapped 	Category 2 	https://github.com/[/issues/5662](https://github.com/mozilla-mobile/focus-android/issues/5662)
> autocomplete arrow tapped 	Category 2 	https://github.com/[/issues/5662](https://github.com/mozilla-mobile/focus-android/issues/5662)
> 
> _6) How long will this data be collected? Choose one of the following:_ Expiry for capturing data is set to "2022-11-01"
> 
> _7) What populations will you measure?_ All release channels and locales.
> 
> _8) If this data collection is default on, what is the opt-out mechanism for users?_ Users can opt out of data collection by disabling Usage and technical data from Settings -> Privacy and security -> Data choices.
> 
> _9) Please provide a general description of how you will analyze this data._ Glean
> 
> _10) Where do you intend to share the results of your analysis?_ Only on Glean, mobile teams, and BD have internal access.
> 
> _12) Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection?_ No.

# Data Review

1) Is there or will there be **documentation** that describes the schema for the ultimate data set in a public, complete, and accurate way?

Yes, through the metrics.yaml and the [Glean Dictionary](https://dictionary.telemetry.mozilla.org/)

2) Is there a control mechanism that allows the user to turn the data collection on and off?

Yes, through the "Send Usage Data" preference in the application settings

3) If the request is for permanent data collection, is there someone who will monitor the data over time?

N/A, collection set to end or be renewed by 2022-11-01

4) Using the **[category system of data types](https://wiki.mozilla.org/Data_Collection)** on the Mozilla wiki, what collection type of data do the requested measurements fall under?

Category 2, Interaction data

5) Is the data collection request for default-on or default-off?

default-on

6) Does the instrumentation include the addition of **any *new* identifiers** (whether anonymous or otherwise; e.g., username, random IDs, etc.  See the appendix for more details)?

No

7) Is the data collection covered by the existing Firefox privacy notice?

Yes

8) Does the data collection use a third-party collection tool?

No

## Result

data-review+

**Review by travis79 — APPROVED** [review]
data-review+

**Review by st3fan — APPROVED** [review]

**Pull request merged**